### PR TITLE
Log errors to their own file for each bot

### PR DESCRIPTION
**Pull request by Quiescent**
- Log the bot error streams to a file per bot (the file is named based on the id and name of the bot.)

### CODE REVIEW

File: runtime/src/runtime.lisp [hunk 1]
```diff
@@ -60,17 +60,29 @@
...
-                                        :bot-name (bot-name bot)
-                                        :bot-definition (bot-definition bot)
-                                        :command (command bot)
-                                        :bot-restarts (+ (bot-restarts bot) 1))))
+  (let* ((bot-id (random-id 10))
+         (bot-name (-> (bot-definition bot) name))
+         (initialised-bot (make-instance 'concrete-bot
+                                         :bot-process (run-bot "bash"
+                                                               (list "-c" (command bot))
+                                                               (open (format nil "~a-~a.err" bot-name bot-id)
```
Comment (HenryS1) [inline runtime/src/runtime.lisp:77]: This is a dangling file handle. I think the bot error file stream should be provided by the the game-runner. An additional stream like the log-stream (perhaps error-stream) could work.

File: runtime/src/runtime.lisp [hunk 2]
```diff
@@ -111,13 +123,20 @@
...
-                    :bot-name (name bot-definition)
-                    :bot-definition bot-definition
-                    :command memory-limited-command)))
+        (let* ((bot-id (random-id 10))
+               (bot-name (name bot-definition))
+               (bot (make-instance
+                     'concrete-bot
+                     :bot-process (run-bot "bash"
+                                           (list "-c" memory-limited-command)
+                                           (open (format nil "~a-~a.err" bot-name bot-id)
```
Comment (HenryS1) [inline runtime/src/runtime.lisp:132]: Same here. I suggest an additional argument to this function. I'm happy to contribute to this change if you'd like.

Comment (Quiescent) [reply runtime/src/runtime.lisp:132]: Nope.  I'm happy to do that myself :)

File: footsoldiers/src/footsoldiers.lisp [hunk 3]
```diff
@@ -714,7 +717,9 @@
 (defun start-game (bot-relative-paths logging-config 
                    &key (current-directory nil)
                      (game-config *default-game-config*)
-                     (game-map *default-game-map*))
+                     (game-map *default-game-map*)
+                     (player1-error-stream *standard-output*)
```
Comment (Quiescent) [inline footsoldiers/src/footsoldiers.lisp:721]: Shouldn't these default to standard error?

File: footsoldiers/src/footsoldiers.lisp [hunk 4]
```diff
@@ -910,6 +921,10 @@
   (write-char char (stream-of stream))
   (write-char char *standard-output*))
 
+(defmethod sb-gray:stream-write-sequence ((stream with-stdout) seq &optional (start 0) end)
+  (write-sequence seq stream start end)
+  (write-sequence seq *standard-output* :start start :end end))
```
Comment (Quiescent) [inline footsoldiers/src/footsoldiers.lisp:926]: I'm not familiar with gray streams.  Why do we write to two streams here?

Comment (HenryS1) [reply footsoldiers/src/footsoldiers.lisp:926]: I write to the file in addition to the output stream

File: footsoldiers/src/footsoldiers.lisp (hunk 3 above)
Comment (HenryS1) [reply footsoldiers/src/footsoldiers.lisp:721]: Yes :)

**Pull request merged**